libcontainer/rootfs_linux.go

```diff
@@ -40,13 +40,13 @@ type mountConfig struct {
 // mountEntry contains mount data specific to a mount point.
 type mountEntry struct {
 	*configs.Mount
-	srcFD   string
+	srcFD   int
 	idmapFD int
```

Comment:
```suggestion
	fd   int
```
I think now that both are type int, we can switch this to just have one `fd`. 

srcFD and idmapFD are mutually exclusive, and we already know based on the mount properties if the fd at hand is a srcFD or an idmapFD.

[Here][link] we could just read "fd" and treat it as an idmapFD. Otherwise, we treat it as a srcFD.

I like the separation of fds, to avoid mistakes from happening, but I think having just one fd and using it appropriately (as all the fds there are mutually exclusive: only one of those fields can be set to something != -1).

What do you think?

I would do that change on top of these commits, though. So it is simpler to review and we can come back to the state the PR is currently at by just removing that commit.

[link]: https://github.com/opencontainers/runc/pull/3939/commits/2e9093faccd36b782788fad77e832179a0a5f692#diff-e0932af06cb46065b2ef5fc89bc2bd9b880c9578e01b89ef4ecaeffb78a8bef0R484-R487

Reply:
I agree, we can call it `fd`, it should unify some code.